app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt

```diff
@@ -220,6 +175,25 @@ class DroneExecutorImpl(
                         data.getMutableMissionPaused().postValue(true)
                     }
 
+    private fun returnTo(ctx: Context, returnLocation: LatLng, altitude: Float, @StringRes msg: Int): Completable {
+
+        val droneInstance = getInstance()
+        val missionPlan = DroneUtils.makeDroneMission(listOf(returnLocation), altitude, null)
+
+        return droneInstance.mission.pauseMission()
+                .doOnComplete {
+                    data.getMutableMissionPaused().postValue(true)
+                    data.getMutableDroneStatus().postValue(SENDING_ORDER)
+                }
+                .andThen(droneInstance.mission.uploadMission(missionPlan)
+                        .andThen(droneInstance.mission.startMission())
+                        .doOnComplete {
+                            data.getMutableDroneStatus().postValue(EXECUTING_MISSION)
+                            data.getMutableMission().postValue(null)
+                            data.getMutableMissionPaused().postValue(false)
+                            ToastHandler.showToastAsync(ctx, msg)
+                        })
```

Comment:
There are extra unnecessary spaces here
```suggestion
                .andThen(droneInstance.mission.startMission())
                .doOnComplete {
                    data.getMutableDroneStatus().postValue(EXECUTING_MISSION)
                    data.getMutableMission().postValue(null)
                    data.getMutableMissionPaused().postValue(false)
                    ToastHandler.showToastAsync(ctx, msg)
                })
```